### PR TITLE
dmarc: start parsing DMARC records

### PR DESCRIPTION
**Pull request by djc**
r? @wez

### CODE REVIEW

File: crates/kumo-dmarc/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let values = [
+            (self.all_pass, '0'),
+            (self.any_pass, '1'),
+            (self.dkim, 'd'),
+            (self.spf, 's'),
+        ];
+
+        let mut first = true;
+        for &(_, ch) in values.iter().filter(|(value, _)| *value) {
```
Comment (wez) [inline crates/kumo-dmarc/src/lib.rs]: I feel like this is a bit easier to grok at first sight:

```suggestion
        for (value, ch) in values {
            if !value {
                continue;
             }
```

File: crates/kumo-dmarc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+        };
+
+        let size = size.trim();
+        if size.is_empty() {
+            return Err(format!("empty size in {s:?}"));
+        }
+
+        let mut power = 0;
+        match size.chars().next_back() {
+            Some('k') => power = 10,
```
Comment (wez) [inline crates/kumo-dmarc/src/lib.rs:177]: heh, I've gone this many years without seeing this particular way of handling this sort of suffix. Neat!

**Review by wez — COMMENTED** [review]
looks good, thank you!
One minor comment, but can merge if you feel strongly about it

File: crates/kumo-dmarc/src/lib.rs (hunk 1 above)
Comment (djc) [reply crates/kumo-dmarc/src/lib.rs]: Fair enough, if I was reviewing this code I might've said the same thing! Will fix it.

**Comment by wez** [comment]
Thanks!

**Pull request merged**